packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -121,6 +120,45 @@ export class InsuredBridgeL2Client {
     });
   }
 
+  async getEventsForBlockSearch(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[][]> {
```

Comment:
Do you think we could make a utility method that has the following interface? (Note: some of the types might be wrong, just wanted to jot an example down)

```ts
async getEventsForMutlipleProviders(web3s: Web3[], contract: Contract, eventName: string, eventParameters: Parameters<Contract["getPastEvents"]>[1]): Promise<EventData[][]> {
```

Or something like that? That way, we can reuse this for any event query?

I think this would allow us to separate out the event-specific logic from the retry/fallback logic.

Reply:
Yep done